cli/npm/resolvers/common.rs

```diff
@@ -95,20 +96,26 @@ pub fn ensure_registry_read_permission(
   permissions: &dyn NodePermissions,
   registry_path: &Path,
   path: &Path,
+  cache: &mut HashMap<PathBuf, Result<(), AnyError>>,
```

Comment:
Instead of storing a result of permission check, maybe we could store a mapping between `registry_path` and `fs.realpath_sync(registry_path)` as well as mapping between `path` and `fs.realpath_sync(path)`?

Reply:
hmmm yea it would make more sense